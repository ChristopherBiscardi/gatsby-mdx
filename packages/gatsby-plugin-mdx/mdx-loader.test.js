const mdxLoader = require("./mdx-loader");
const prettier = require("prettier");

const fixtures = [
  [
    "one.mdx",
    {
      internal: { type: "File" },
      sourceInstanceName: "webpack-test-fixtures",
      absolutePath: "/fake/one.mdx"
    },
    "# some heading"
  ],
  [
    "two.mdx",
    {
      internal: { type: "File" },
      sourceInstanceName: "webpack-test-fixtures",
      absolutePath: "/fake/two.mdx"
    },
    `# Two things

some paragraph content

**bold**`
  ]
];

describe("mdx-loader", () => {
  expect.addSnapshotSerializer({
    print(val, serialize) {
      return prettier.format(val, { parser: "babylon" });
    },
    test() {
      return true;
    }
  });
  test.each(fixtures)(
    "snapshot %s",
    async (filename, fakeGatsbyNode, content) => {
      const loader = mdxLoader.bind({
        async() {
          return (err, result) => {
            expect(err).toBeNull();
            expect(result).toMatchSnapshot();
          };
        },
        query: {
          getNodes() {
            return fixtures.map(([filename, node, ...other]) => node);
          },
          pluginOptions: {}
        },
        resourcePath: fakeGatsbyNode.absolutePath
      });
      await loader(content);
    }
  );
});
