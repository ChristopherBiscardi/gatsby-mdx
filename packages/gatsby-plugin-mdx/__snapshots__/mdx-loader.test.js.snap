// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mdx-loader snapshot one.mdx 1`] = `
import React from "react";
import { MDXTag } from "@mdx-js/tag";

export default ({ components, ...props }) => (
  <MDXTag name="wrapper" components={components}>
    <MDXTag name="h1" components={components}>{\`some heading\`}</MDXTag>
  </MDXTag>
);

export const _frontmatter = {};

`;

exports[`mdx-loader snapshot two.mdx 1`] = `
import React from "react";
import { MDXTag } from "@mdx-js/tag";

export default ({ components, ...props }) => (
  <MDXTag name="wrapper" components={components}>
    <MDXTag name="h1" components={components}>{\`Two things\`}</MDXTag>
    <MDXTag name="p" components={components}>{\`some paragraph content\`}</MDXTag>
    <MDXTag name="p" components={components}>
      <MDXTag
        name="strong"
        components={components}
        parentName="p"
      >{\`bold\`}</MDXTag>
    </MDXTag>
  </MDXTag>
);

export const _frontmatter = {};

`;
